src/System.Net.Http/tests/FunctionalTests/LoopbackServer.cs

```diff
@@ -0,0 +1,112 @@
+        }
+
+        public enum TransferError
+        {
+            None = 0,
+            ContentLengthTooLarge,
+            ChunkSizeTooLarge,
+            MissingChunkTerminator
+        }
+
+        public static Task StartServer(
+            TransferType transferType,
+            TransferError transferError,
+            out IPEndPoint serverEndPoint)
+        {
+            var server = new TcpListener(IPAddress.Loopback, 0);
+            Task serverTask = ((Func<Task>)async delegate {
+                server.Start();
+                using (var client = await server.AcceptSocketAsync())
+                using (var stream = new NetworkStream(client))
+                using (var reader = new StreamReader(stream, Encoding.ASCII))
+                {
+                    // Read past request headers.
+                    string line;
+                    while (!string.IsNullOrEmpty(line = reader.ReadLine())) ;
+
+                    // Determine response transfer headers.
+                    string transferHeader = null;
+                    string content = "This is some response content.";
+                    if (transferType == TransferType.ContentLength)
+                    {
+                        if (transferError == TransferError.ContentLengthTooLarge)
+                        {
+                            transferHeader = $"Content-Length: {content.Length + 42}\r\n";
+                        }
+                        else
+                        {
+                            transferHeader = $"Content-Length: {content.Length}\r\n";
+                        }
+                    }
+                    else if (transferType == TransferType.Chunked)
+                    {
+                        transferHeader = "Transfer-Encoding: chunked\r\n";
+                    }
+
+                    // Write response.
+                    using (var writer = new StreamWriter(stream, Encoding.ASCII))
+                    {
+                        writer.Write("HTTP/1.1 200 OK\r\n");
+                        writer.Write($"Date: {DateTimeOffset.UtcNow:R}\r\n");
+                        writer.Write("Content-Type: text/plain\r\n");
+
+                        if (!string.IsNullOrEmpty(transferHeader))
+                        {
+                            writer.Write(transferHeader);
+                        }
+
+                        writer.Write("\r\n");
+                        if (transferType == TransferType.Chunked)
+                        {
+                            string chunkSizeInHex;
+                            if (transferError == TransferError.ChunkSizeTooLarge)
+                            {
+                                chunkSizeInHex = string.Format("{0:x}\r\n", content.Length + 42);
+                            }
+                            else
+                            {
+                                chunkSizeInHex = string.Format("{0:x}\r\n", content.Length);
+                            }
```

Comment:
This is fine, but it could also be simplified to have a ternary conditional for just the one argument that differs.